flypipe/node.py

```diff
@@ -369,8 +328,49 @@ def _get_consolidated_output_schema(cls, output_schema, output_columns):
             schema = None
         return schema
 
+    def process_transformation_with_cache(
+            self, runnable_node, run_context: RunContext
+    ):
+
+        result = None
+        node_transformation = runnable_node["transformation"]
+        ran_transformation = False
+
+        if runnable_node["status"] == RunStatus.CACHED and \
+                runnable_node["node_run_context"].cache_context.exists_cache_to_load:
+            result = runnable_node["node_run_context"].cache_context.read()
+        else:
+            ran_transformation = True
+            dependency_values = runnable_node["transformation"].get_node_inputs(
+                run_context
+            )
+
+            result = node_transformation.process_transformation(
+                run_context.spark,
+                runnable_node["output_columns"],
+                runnable_node["node_run_context"],
+                **dependency_values,
+            )
+
+        run_context.update_node_results(

```

Comment:
This function is doing two things- updating the run context node results and boxing the result dataframe in a wrapper. It  should not handle wrapping the result as there's no inherent link between that and the primary task of updating the run context.

Reply:
run context already holds provided inputs. we need a single place to hold the node results.
What solution do you recommend?

Reply:
Hmm was going to suggest moving the boxing here but the spark session is in the run context. Just leave as is then.